pkg/k8s/backend.go

```diff
@@ -367,53 +370,57 @@ func translatePeer(peer *v1alpha1.Peer) *mesh.Peer {
 	if peer == nil {
 		return nil
 	}
-	var aips []*net.IPNet
+	var aips []net.IPNet
 	for _, aip := range peer.Spec.AllowedIPs {
 		aip := normalizeIP(aip)
 		// Skip any invalid IPs.
 		if aip == nil {
 			continue
 		}
-		aips = append(aips, aip)
+		aips = append(aips, *aip)
 	}
-	var endpoint *wireguard.Endpoint
+	var endpoint *net.UDPAddr
+	var addr string
 	if peer.Spec.Endpoint != nil {
 		ip := net.ParseIP(peer.Spec.Endpoint.IP)
 		if ip4 := ip.To4(); ip4 != nil {
 			ip = ip4
 		} else {
 			ip = ip.To16()
 		}
-		if peer.Spec.Endpoint.Port > 0 && (ip != nil || peer.Spec.Endpoint.DNS != "") {
-			endpoint = &wireguard.Endpoint{
-				DNSOrIP: wireguard.DNSOrIP{
-					DNS: peer.Spec.Endpoint.DNS,
-					IP:  ip,
-				},
-				Port: peer.Spec.Endpoint.Port,
+		if peer.Spec.Endpoint.Port > 0 {
+			if ip != nil {
+				endpoint = &net.UDPAddr{IP: ip, Port: int(peer.Spec.Endpoint.Port)}
+			}
+			if peer.Spec.Endpoint.DNS != "" {
+				addr = fmt.Sprintf("%s:%d", peer.Spec.Endpoint.DNS, peer.Spec.Endpoint.Port)
 			}
 		}
 	}
-	var key []byte
-	if len(peer.Spec.PublicKey) > 0 {
-		key = []byte(peer.Spec.PublicKey)
-	}
-	var psk []byte
-	if len(peer.Spec.PresharedKey) > 0 {
-		psk = []byte(peer.Spec.PresharedKey)
+
+	key, _ := wgtypes.ParseKey(peer.Spec.PublicKey)
```

Comment:
What about this error?

Reply:
Should we log it? We were ignoring it before (or we didn't know if this was a valid key)

Reply:
Yeah i think that's a good idea